juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,142 @@
-                # Sometimes the court includes "To be released..."
-                # sections, without links, which we skip here
+                url = fromstring(case["documentContent"]).xpath(
+                    ".//div[@class='document__detail__title']//a/@href"
+                )[0]
+            except:
+                logger.info(f"Cannot find file for case: {case}")
                 continue
-            dates.extend([case_date] * len(element.xpath(sub_path)))
-        return dates
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
-
-    def _get_docket_numbers(self):
-        return [pair["docket"] for pair in self.docket_name_pairs]
-
-    def _get_anchor_docket_name_pairs(self):
-        """The court has some ugly HTML practices that we need to handle.
-
-        Most anchor links include single line strings with a single docket
-        number and a single case name.  However, there are two other formats
-        we've seen and must work around.
-
-        (CASE 1)
-        The anchor has multiple lines broken with <br> tag(s), and each
-        line contains "<docket> <name>". In this case we need to combine
-        the docket numbers and name strings respectively.
-        [EXAMPLE: February 18, 2016 in nh_example_2.html]
-
-        (CASE 2)
-        The anchor has multiple lines broken with <br> tag(s), and the
-        second line is a continuation of a long case name started on the first
-        line.  So, the second line does not lead with a docket number, thus
-        this line's string should be glued onto the <name> substring extracted
-        from the previous line.
-        [EXAMPLE: September 18, 2018 in nh_example_6.html]
-        """
-        pairs = []
-        for anchor in self.html.xpath(self.link_path):
-            i = 0
-            dockets = []
-            name_substrings = []
-            text_anchor = anchor.text_content()
-            text_clean = text_anchor.replace("\n", "")
+            try:
+                docket_name = re.search(name_re, case["documentName"])
+                docket = docket_name.group("docket").strip()
+                name = docket_name.group("name").strip().lstrip(",.").strip()
+                docket_name_sec = re.search(name_sec_re, name)
+                if docket_name_sec:
+                    name_fir = (
+                        docket_name_sec.group("name_1")
+                        .strip()
+                        .rstrip(",")
+                        .strip()
+                    )
+                    docket_sec = (
+                        docket_name_sec.group("docket_2")
+                        .strip()
+                        .lstrip(",")
+                        .strip()
+                    )
+                    name_sec = (
+                        docket_name_sec.group("name_2")
+                        .strip()
+                        .lstrip(",.")
+                        .strip()
+                    )
```

Comment:
same here... im reading this and I don't understand what docket_name_sec is

Reply:
Moved to a function for simplification